apps/opentelemetry_api/lib/open_telemetry/tracer.ex

```diff
@@ -126,10 +126,9 @@ defmodule OpenTelemetry.Tracer do
 
   The Span in the current Context has its `is_recording` set to `false`.
   """
-  def end_span() do
-    non_recording_span = :otel_span.end_span(:otel_tracer.current_span_ctx())
-    _ = :otel_tracer.set_current_span(non_recording_span)
-    non_recording_span
+  def end_span(timestamp \\ :undefined) do
+    non_recording_span = :otel_span.end_span(:otel_tracer.current_span_ctx(), timestamp)
+    :otel_tracer.set_current_span(non_recording_span)
```

Comment:
```suggestion
    _ = :otel_tracer.set_current_span(non_recording_span)
    non_recording_span
```

No idea, but maybe dialyxer doesn't like this new code even though it should be considered the same.